src/app/services/restaurant.service.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { HttpResponse } from '../shared/Models/http';
+import { url } from '../shared/constants';
+import { map } from 'rxjs';
+import { Restaurant } from '../shared/Models/restaurant';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RestaurantService {
+  private restaurantIdUrl = url.restaurantIdUrl;
+  private restaurantListUrl = url.restaurantListUrl;
+
+  constructor(private http: HttpClient) {}
+
+  newRestaurant(restaurant: Restaurant) {
+    let body = { selectedRestaurant: restaurant };
+    return this.http.post<HttpResponse>(`${this.restaurantIdUrl}`, body).pipe(
+      map((response) => {
+        if (response.isSuccess) {
+          return response.payLoad;
+        }
+      })
+    );
+  }
+
+  getRestaurant() {
```

Comment:
This method is used to fetch all restaurants so let's make the naming in plural. Especially since we will have a method that will fetch only one restaurant later in the project.
```suggestion
  getRestaurants() {
```